public/app/core/services/context_srv.ts

```diff
@@ -61,6 +73,20 @@ export class ContextSrv {
     return this.user.orgRole === role;
   }
 
+  // Checks whether user has required permission
+  hasPermission(action: AccessControlAction, scope?: AccessControlScope): boolean {
+    // Fallback if access control disabled
+    if (!config.featureToggles['accesscontrol']) {
+      return true;
+    }
+
+    return !!(
+      this.user.permissions &&
+      this.user.permissions[action] &&
+      (scope ? this.user.permissions[action][scope] : true)
+    );
```

Comment:
```suggestion
    return !!(this.user.permissions?.[action] && (scope ? this.user.permissions[action][scope] : true));
```

Reply:
This is how prettier formats this.

Reply:
This is not about formatting, you can remove an extra `this.user.permissions` check with the optional chaining operator.

Reply:
Got it. Did it before, but had some linter errors. Will try again.